Automattic/jetpack
# JS Components: Convert AutomatticBylineLogo component to TypeScript

#### Changes proposed in this Pull Request:
* This PR converts `AutomatticBylineLogo` component to TypeScript and cleans up the propTypes.
* It also converts the `AutomatticBylineLogo` stories and unit tests to TS
* Adds more test cases for `AutomatticBylineLogo` unit tests

#### Other information:


#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:
* Checkout the PR
* Run `cd projects/js-packages/storybook && pnpm run storybook:dev`
* Confirm that all the stories of `AutomatticBylineLogo` component work as expected
* Check the consumers of `AutomatticBylineLogo` and related components and confirm that they work as expected
* Check for any regression

<img width="1748" alt="Screenshot 2022-07-04 at 2 05 56 PM" src="https://user-images.githubusercontent.com/18226415/177116210-0f160d1f-e230-429a-b02d-20d222b33fcf.png">


---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202276314263996
